src/test/kotlin/io/bazel/kotlin/builder/tasks/jvm/KotlinBuilderJvmJdepsTest.kt

```diff
@@ -1385,7 +1621,7 @@ class KotlinBuilderJvmJdepsTest(private val enableK2Compiler: Boolean) {
 
     assertExplicit(jdeps).contains(depWithFunction.singleCompileJar())
     assertImplicit(jdeps).contains(depWithReturnType.singleCompileJar())
-    assertImplicit(jdeps).doesNotContain(depWithReturnTypesSuperType)
+    assertImplicit(jdeps).contains(depWithReturnTypesSuperType.singleCompileJar())
```

Comment:
Note that the prior assertion always succeeded as it was missing `.singleCompileJar()`
When .singleCompileJar() is correctly added, the assertion fails. 
We can confirm that depWithReturnTypesSuperType is needed for compilation by commenting out:
`c.addTransitiveDependencies(depWithReturnTypesSuperType)`
and then seeing the compilation fail with:
```
error: supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
    class something.SomeType, unresolved supertypes: something.SomeSuperType
```
Because the super type is actually needed, we've changed the assertion to assert that it is an implicit dep.